samples/star-wars-api/HttpHandlers.fs

```diff
@@ -62,48 +103,47 @@ module HttpHandlers =
 
             let request = ctx.Request
 
-            let hasData () =
-                if request.Body.CanSeek then
-                    request.Body.Length > 0L
-                else
-                    request.EnableBuffering()
-                    let body = request.Body
-                    let buffer = Array.zeroCreate 1
-                    let bytesRead = body.Read(buffer, 0, 1)
-                    body.Seek(0, SeekOrigin.Begin) |> ignore
-                    bytesRead > 0
-
-            if (request.Method = HttpMethods.Get || not <| hasData ())
+            let hasData () = task {
+            if request.Body.CanSeek then
+                return (request.Body.Length > 0L)
+            else
+                request.EnableBuffering()
+                let body = request.Body
+                let buffer = Array.zeroCreate 1
+                let! bytesRead = body.ReadAsync(buffer, 0, 1)
+                body.Seek(0, SeekOrigin.Begin) |> ignore
+                return (bytesRead > 0)
+            }
+            if (request.Method = HttpMethods.Get)
             then
+                logger.LogInformation("Executing GraphQL introspection query")
                 let! result = Schema.executor.AsyncExecute (Introspection.IntrospectionQuery)
-                printfn "Result metadata: %A" result.Metadata
-                let response = toResponse result.DocumentId result.Content
+                let response = result |> toResponse
                 return Results.Ok response
             else
-
+            let! hd = hasData()
+            if (not hd)
+            then
+                logger.LogInformation("Executing GraphQL introspection query")
+                let! result = Schema.executor.AsyncExecute (Introspection.IntrospectionQuery)
+                let response = result |> toResponse
+                return Results.Ok response
+            else
```

Comment:
Let's modify this a bit.
As we discovered that introspection query can come in a body too, we can add `checkIsIntrospectionQuery` function and then if on it.
``` F#
[<Struct>]
type OperationType =
	| IntrospectionQuery of string option
	| OperationQuery of GQLRequestContent

let checkIsNotIntrospectionQuery = task {
    if isGet then IntrospectionQuery ValueNone
    else
    let! hasbody = checkIfHasBody()
    if not hasBody then IntrospectionQuery ValueNone
    else
    let! request = ctx.BindJsonAsync<GQLRequestContent>()
    // This needs to be verified and can be omitted in the current PR
    if Introspection.IntrospectionQuery.Contains request.Query then IntrospectionQuery request.Query
    else OperationQuery request
}

let executeIntrospectionQuery query = task {
	let! result =
		match query with
		| ValueNone ->
			logger.LogInformation("Executing default GraphQL introspection query")
			Schema.executor.AsyncExecute (Introspection.IntrospectionQuery)
		| ValueSome query ->
			logger.LogInformation($"Executing GraphQL introspection query:{Environment.NewLine}{query}", query)
			Schema.executor.AsyncExecute (query)
		let response = result |> toResponse
		return Results.Ok response
}

let executeOperation request = task {
	...
}

match! checkIsNotIntrospectionQuery with
| IntrospectionQuery query -> return! executeIntrospectionQuery query 
| OperationQuery request -> return! executeOperation request
```